swarm/src/lib.rs

```diff
@@ -1527,6 +1534,47 @@ where
         }
     }
 
+    /// Sets executor to the `wasm` executor.
+    /// Background tasks will be executed by the browser on the next micro-tick.
+    ///
+    /// Spawning a task is similar too:
+    /// ```typescript
+    /// function spawn(task: () => Promise<void>) {
+    ///     task()
+    /// }
+    /// ```
+    #[cfg(feature = "wasm-bindgen")]
+    pub fn with_wasm_executor(
+        transport: transport::Boxed<(PeerId, StreamMuxerBox)>,
+        behaviour: TBehaviour,
+        local_peer_id: PeerId,
+    ) -> Self {
+        Self::with_executor(
+            transport,
+            behaviour,
+            local_peer_id,
+            crate::executor::WasmBindgenExecutor,
+        )
+    }
+
+    /// Sets executor to a threadpool executor.
+    pub fn with_threadpool_executor(
+        transport: transport::Boxed<(PeerId, StreamMuxerBox)>,
+        behaviour: TBehaviour,
+        local_peer_id: PeerId,
+    ) -> Self {
+        match ThreadPoolBuilder::new()
+            .name_prefix("libp2p-swarm-task-")
+            .create()
+        {
+            Ok(tp) => Self::with_executor(transport, behaviour, local_peer_id, tp),
+            Err(err) => {
+                log::warn!("Failed to create executor thread pool: {:?}", err);
+                Self::without_executor(transport, behaviour, local_peer_id)
+            }
+        }
+    }
```

Comment:
Let's not introduce this one. See https://github.com/libp2p/rust-libp2p/issues/3107.

Reply:
Actually, I cannot simply not introduce it. 
```bash
error[E0599]: no function or associated item named `with_threadpool_executor` found for struct `SwarmBuilder` in the current scope
  --> examples/file-sharing/src/network.rs:50:31
   |
50 |     let swarm = SwarmBuilder::with_threadpool_executor(
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^
   |                               |
   |                               function or associated item not found in `SwarmBuilder<_>`
   |                               help: there is an associated function with a similar name: `with_executor`
```
So I am keep it for now and then it will be removed as part of #3107?

Reply:
I'd like to not introduce code only to deprecate it later again. Would you mind porting the file-sharing example to a different executor?